### PR TITLE
Add postAsyncGetter to the List call

### PR DESCRIPTION
**Pull request by smithab**
Please have a look at our [guidelines for contributions](https://github.com/Microsoft/AppCenter-SDK-Android/blob/develop/CONTRIBUTING.md) and consider the following before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [ ] Are tests passing locally?
* [ ] Are the files formatted correctly?
* [ ] Did you add unit tests?
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?
* [ ] Did you check UI tests on the sample app? They are not executed on CI.

## Description

A few sentences describing the overall goals of the pull request.

## Related PRs or issues

List related PRs and other issues.

## Misc

Add what's missing, notes on what you tested, additional thoughts or questions.

### CODE REVIEW

File: sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/Storage.java [hunk 1]
```diff
@@ -369,8 +369,8 @@ protected String getLoggerTag() {
         }
         postAsyncGetter(new Runnable() {
 
-            @Override
-            public void run() {
+                @Override
```
Comment (YiMysty) [inline sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/Storage.java]: a line :)

File: apps/sasquatch/src/projectDependency/java/com/microsoft/appcenter/sasquatch/activities/StorageActivity.java [hunk 2]
```diff
@@ -74,26 +74,32 @@
...
-        public void accept(PaginatedDocuments<TestDocument> documents) {            
+        public void accept(PaginatedDocuments<TestDocument> documents) {
             mAppDocumentsLoading = false;
             if (!mUserDocumentsLoading) {
-               hideProgress();
+                hideProgress();
             }
             mCurrentAppDocuments = documents;
-            updateAppDocument(documents.getCurrentPage().getItems());
+
```
Comment (guperrot) [inline apps/sasquatch/src/projectDependency/java/com/microsoft/appcenter/sasquatch/activities/StorageActivity.java]: Screen Shot 2019-04-19 at 4 37 28 PM

Please always commit using those options in Android Studio, this code is not formatted correctly.

Also we don't put blank lines without a comment.

File: sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/Storage.java (hunk 1 above)
Comment (YiMysty) [reply sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/Storage.java]: extra space?

File: apps/sasquatch/src/projectDependency/java/com/microsoft/appcenter/sasquatch/activities/StorageActivity.java [hunk 3]
```diff
@@ -74,26 +74,32 @@
...
         @Override
-        public void accept(PaginatedDocuments<Map> documents) {            
+        public void accept(PaginatedDocuments<Map> documents) {
             mUserDocumentsLoading = false;
             if (!mAppDocumentsLoading) {
                 hideProgress();
             }
             mCurrentUserDocuments = documents;
-            updateUserDocuments(documents.getCurrentPage().getItems());
+
```
Comment (guperrot) [inline apps/sasquatch/src/projectDependency/java/com/microsoft/appcenter/sasquatch/activities/StorageActivity.java]: same here

File: sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/Storage.java [hunk 4]
```diff
@@ -558,21 +558,28 @@ public void onCallFailed(Exception e) {
         if (isInvalidPartitionWhenDocuments(partition, result)) {
             return result;
         }
-        getTokenAndCallCosmosDbApi(
-                partition,
-                result,
-                new TokenExchangeServiceCallback(mTokenManager) {
```
Comment (guperrot) [inline sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/Storage.java]:
```suggestion
```

File: apps/sasquatch/src/projectDependency/java/com/microsoft/appcenter/sasquatch/activities/StorageActivity.java [hunk 5]
```diff
@@ -74,26 +74,32 @@
...
+        public void accept(PaginatedDocuments<TestDocument> documents) {
             mAppDocumentsLoading = false;
             if (!mUserDocumentsLoading) {
-               hideProgress();
+                hideProgress();
             }
             mCurrentAppDocuments = documents;
-            updateAppDocument(documents.getCurrentPage().getItems());
+
+            if(documents != null && documents.getCurrentPage() != null){
```
Comment (YiMysty) [inline apps/sasquatch/src/projectDependency/java/com/microsoft/appcenter/sasquatch/activities/StorageActivity.java]: extra line? :D

File: sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/Storage.java [hunk 6]
```diff
@@ -558,21 +557,26 @@ public void onCallFailed(Exception e) {
...
+        postAsyncGetter(new Runnable() {
+            @Override
```
Comment (dhei) [inline sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/Storage.java]: Nit: add a new line break before `@Override` - our conventions it to have an empty line before Java annotation.
```suggestion

            @Override
```

File: sdk/appcenter-storage/src/test/java/com/microsoft/appcenter/storage/StorageTest.java [hunk 7]
```diff
@@ -188,6 +188,13 @@ public ServiceCall answer(InvocationOnMock invocation) {
         assertFalse(docs.hasNextPage());
         assertEquals(1, docs.getCurrentPage().getItems().size());
         assertEquals(docs.getCurrentPage().getItems().get(0).getDocument().test, documents.get(0).getDocument().test);
+
+        /* Disable the Storage module */
```
Comment (guperrot) [inline sdk/appcenter-storage/src/test/java/com/microsoft/appcenter/storage/StorageTest.java]:
```suggestion
        /* Disable the Storage module. */
```

**Pull request merged**